File: interviews/leetcode/hash-tables-and-sets/2352. Equal Row and Column Pairs/solution.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:

        n = len(grid)
        # Step 1: Convert rows to tuples
        rows = [tuple(grid[i]) for i in range(n)]

        # Step 2: Convert columns to tuples
        columns = [tuple(grid[i][j] for i in range(n)) for j in range(n)]

        # Step 3: Count equal pairs of rows and columns
        count = 0
        for row in rows:
            for col in columns:
                if row == col:
                    count += 1

        return count

File: interviews/leetcode/hash-tables-and-sets/2352. Equal Row and Column Pairs/solution.go
func equalPairs(grid [][]int) int {
	n := len(grid)
	count := 0

	// Convert rows to tuples (slice)
	rows := make([][]int, n)
	for i := 0; i < n; i++ {
		rows[i] = make([]int, n)
		copy(rows[i], grid[i])
	}

	// Convert columns to slices
	columns := make([][]int, n)
	for j := 0; j < n; j++ {
		columns[j] = make([]int, n)
		for i := 0; i < n; i++ {
			columns[j][i] = grid[i][j]
		}
	}

	// Compare rows and columns
	for _, row := range rows {
		for _, col := range columns {
			equal := true
			for k := 0; k < n; k++ {
				if row[k] != col[k] {
					equal = false
					break
				}
			}
			if equal {
				count++
			}
		}
	}

	return count
}

/*
func main() {
    grid := [][]int{
        {3, 2, 1},
        {1, 2, 3},
        {3, 2, 1},
    }
    fmt.Println(equalPairs(grid))  // Output: 2
}
*/
